frontend/src/App.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import FrontScreen from './Pages/FrontScreen';
+import NavBar from './NavBar';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      "type": null,
+      "latitude": null,
+      "longitude": null
+    }
+    this.setLocation = this.setLocation.bind(this)
+  }
+
+  setType(type) {
+    this.setState({
+      type: type,
+      latitude: this.state.latitude,
+      longitude: this.state.longitude
```

Comment:
I think setState does a merge, can you check the behavior 

this.setState({
      type: type,
      latitude: this.state.latitude,
      longitude: this.state.longitude
})


this.setState({type: type})

are equivalent

Reply:
Tip: In ES6, you can just write `this.setState({type})`. If you do not specify the property name, variable's name is picked as the property name